imxrt-hal/src/can/mod.rs

```diff
@@ -0,0 +1,1040 @@
+        if let Some(rximr) = rximr {
+            unsafe { core::ptr::write_volatile((mailbox_rximr_addr) as *mut u32, rximr) };
+        }
+    }
+
+    fn write_mailbox_idflt_tab(&self, mailbox_number: u8, idflt_tab: Option<u32>) {
+        let mailbox_idflt_tab_addr = self.mailbox_number_to_idflt_tab_address(mailbox_number);
+        if let Some(idflt_tab) = idflt_tab {
+            unsafe { core::ptr::write_volatile((mailbox_idflt_tab_addr) as *mut u32, idflt_tab) };
+        }
+    }
+
+    fn enable_mailbox_interrupt(&mut self, mailbox_number: u8, enabled: bool) {
+        /* mailbox not available */
+        if mailbox_number < self.mailbox_offset() {
+            return;
+        }
+        if enabled {
+            /* enable mailbox interrupt */
+            self.write_imask_bit(mailbox_number, true);
+            return;
+        } else {
+            match self.read_mailbox(mailbox_number) {
+                Some(d) => {
+                    if (d.frame.code.to_code_reg() & 0x0F000000) >> 3 != 0 {
+                        /* transmit interrupt keeper */
+                        self.write_imask_bit(mailbox_number, true);
+                        return;
+                    }
+                }
+                _ => {}
+            }
+        }
+        /* disable mailbox interrupt */
+        self.write_imask_bit(mailbox_number, false);
+        return;
+    }
+
+    pub fn read_mailboxes(&mut self) -> Option<MailboxData> {
+        let mut iflag: u64;
+        let mut cycle_limit: u8 = 3;
+        let offset = self.mailbox_offset();
+        let max_mailbox = self.get_max_mailbox();
+        while self._mailbox_reader_index <= max_mailbox {
+            iflag = self.read_iflag();
+            if iflag != 0 && (self._mailbox_reader_index >= (64 - iflag.leading_zeros() as u8)) {
+                /* break from MSB's if unset, add 1 to prevent undefined behaviour in clz for 0 check */
+                self._mailbox_reader_index = self.mailbox_offset();
+                cycle_limit -= 1;
+                if cycle_limit == 0 {
+                    return None;
+                }
+            }
+            if self.fifo_enabled() {
+                /* FIFO is enabled, get only remaining RX (if any) */
+                if self._mailbox_reader_index < offset {
+                    /* go back to position end of fifo+filter region */
+                    self._mailbox_reader_index = offset;
+                }
+            }
+            if self._mailbox_reader_index >= max_mailbox {
+                self._mailbox_reader_index = self.mailbox_offset();
+                cycle_limit -= 1;
+                if cycle_limit == 0 {
+                    return None;
+                }
+            }
+            if (self.read_imask() & (1_u64 << self._mailbox_reader_index)) != 0 {
+                self._mailbox_reader_index += 1;
+                continue; /* don't read interrupt enabled mailboxes */
```

Comment:
> don't read interrupt enabled mailboxes

Is this documented somewhere for the user? Although `handle_interrupt` also returns a `MailboxData`, it might not be obvious that `handle_interrupt` is XOR `read_mailboxes`.